### PR TITLE
Assorted improvements to the bom

### PR DESCRIPTION
**Pull request by gsmet**
* https://hibernate.atlassian.net/browse/OGM-1138
 * https://hibernate.atlassian.net/browse/OGM-1140
 * https://hibernate.atlassian.net/browse/OGM-1141: there's more to it as we weren't compatible with MongoDB 3.2 so the commit also contains the fixes for that
 * https://hibernate.atlassian.net/browse/OGM-1142
 * https://hibernate.atlassian.net/browse/OGM-1137

### CODE REVIEW

File: bom/pom.xml [hunk 1]
```diff
@@ -51,9 +51,9 @@
         <ehcacheVersion>2.6.11</ehcacheVersion>
 
         <!-- MongoDB -->
-        <mongodbDriverVersion>3.2.2</mongodbDriverVersion>
-        <mongodbVersion>3.0.2</mongodbVersion>
-        <fongodbVersion>2.0.6</fongodbVersion>
+        <mongodbDriverVersion>3.3.0</mongodbDriverVersion>
+        <mongodbVersion>3.2.8</mongodbVersion>
```
Comment (gunnarmorling) [inline bom/pom.xml]: Would it still work with 3.0 or is 3.2 required by OGM now?

File: bom/pom.xml [hunk 2]
```diff
@@ -21,55 +21,70 @@
...
+        <hibernateHqlParserVersion>1.3.0.Final</hibernateHqlParserVersion>
+        <parboiledVersion>1.1.7</parboiledVersion> <!-- needs to be aligned with Neo4j version but is also used for MongoDB -->
+        <asmVersion>5.0.3</asmVersion> <!-- used for Parboiled and Neo4j -->
+
+        <!-- Ehcache -->
+        <ehcacheVersion>2.6.11</ehcacheVersion>
+
+        <!-- MongoDB -->
+        <mongodbDriverVersion>3.3.0</mongodbDriverVersion>
+        <mongodbVersion>3.2.8</mongodbVersion>
```
Comment (gunnarmorling) [inline bom/pom.xml]: Would it still work with 3.0 or is 3.2 required by OGM now?

**Comment by gunnarmorling** [comment]
Looks good apart from my one question.

File: bom/pom.xml (hunk 2 above)
Comment (gsmet) [inline bom/pom.xml]: It still works with 3.0. The only test failing is the one testing the index as they changed the textIndexVersion from 2 to 3. Other than that, there are no incompatible changes.

**Comment by gunnarmorling** [comment]
The rebase doesn't cleanly apply. Could you do it and then merge yourself, @gsmet?

**Comment by gsmet** [comment]
Rebased and merged! Thanks for the review.

**Pull request closed without merging**